bot/exts/pride/pride_leader.py

```diff
@@ -0,0 +1,115 @@
+import json
+import logging
+import random
+from pathlib import Path
+from typing import Optional
+
+import discord
+from discord.ext import commands
+from fuzzywuzzy import fuzz
+
+from bot import bot
+from bot import constants
+
+log = logging.getLogger(__name__)
+
+PRIDE_RESOURCE = json.loads(Path("bot/resources/pride/prideleader.json").read_text("utf8"))
+MINIMUM_FUZZ_RATIO = 40
+
+
+class PrideLeader(commands.Cog):
+    """Gives information about Pride Leaders."""
+
+    def __init__(self, bot: bot.Bot):
+        self.bot = bot
+
+    def invalid_embed_generate(self, pride_leader: str) -> discord.Embed:
+        """
+        Generates Invalid Embed.
+
+        The invalid embed contains a list of closely matched names of the invalid pride
+        leader the user gave. If no closely matched names are found it would list all
+        the available pride leader names.
+
+        Wikipedia is a useful place to learn about pride leaders and we don't have all
+        the pride leaders, so the bot would add a field containing the wikipedia
+        command to execute.
+        """
+        embed = discord.Embed(
+            color=constants.Colours.soft_red
+        )
+        valid_names = []
+        pride_leader = pride_leader.title()
+        for name in PRIDE_RESOURCE:
+            if fuzz.ratio(pride_leader, name) >= MINIMUM_FUZZ_RATIO:
+                valid_names.append(name)
+
+        if not valid_names:
+            valid_names = ", ".join(PRIDE_RESOURCE)
+            error_msg = "Sorry your input didn't match any stored names, here is a list of available names:"
+        else:
+            valid_names = "\n".join(valid_names)
+            error_msg = "Did you mean?"
+
+        embed.description = f"{error_msg}\n```{valid_names}```"
+        embed.set_footer(text="To add more pride leaders, feel free to open a pull request!")
+
+        return embed
+
+    def embed_builder(self, leader_name: str) -> discord.Embed:
```

Comment:
I recommend taking in the leader `dict` instead of the name. Since the input validation gets the dictionary from the resource, you don't have to access it twice. Choosing a random dictionary (in `pride_leader()`) is also simpler than choosing a random name out of a list built from dictionary keys, then getting the same dictionary using the name.